File: python/candy.py
def merge (v, begin, end):
	middle = (begin + end) // 2
	sv = []
	i = 0
	j = middle + 1
	while i <= middle and j <= end:
		if v[i] < v[j]:
			sv.append (v[i])
			i += 1
		else:
			sv.appemd (v[j])
			j += 1
	while i <= middle:
		sv.append (v[i])
		i += 1
	while j <= end:
		sv.append (v[j])
		j += 1
	for i in range (begin, end + 1):
		v[i] = sv[i - begin]




def sort (v, begin, end):
	if end > begin:
		middle = (begin + end) // 2
		sort (v, begin, middle)
		sort (v, middle + 1, end)
		merge (v, begin, end)

def allDifferent (v):
	for i in range (0, len (v) - 1):
		if v[i] != v[i+1]:
			return True
	return False




while True:
	N = int (raw_input())
	if N == -1:
		break
	else:
		packets = []
		for i in range (0, N):
			packets.append (int (raw_input()))
		if sum (packets) % len (packets):
			print -1
		else:
			result = 0
			mean = sum (packets) / len (packets)
			i = 0
			j = len (packets) - 1
			while allDifferent (packets):
				while packets[i] != mean:
					if packets[j] > mean:
						packets[i] += 1
						packets[j] -= 1
						result += 1
					else:
						j -= 1
				i += 1
			print result

File: cplusplus/candy.cpp
#include <iostream>
#include <vector>


using namespace std;


int sum (vector <int> &vec)
{
	int sum = 0;
	for (int i = 0; i < vec.size(); ++i)
	{
		sum += vec[i];
	}
	return sum;
}


void sort (vector <int> &vec)
{
	for (int i = 0; i < vec.size(); ++i)
	{
		for (int j = 0; j < vec.size() - 1 - i; ++j)
		{
			if (vec[j] > vec[j+1])
			{
				int temp = vec[j];
				vec[j] = vec[j+1];
				vec[j+1] = temp;
			}
		}
	}
}


bool allDifferent (vector <int> &vec)
{
	for (int i = 0; i < vec.size() - 1; ++i)
	{
		if (vec[i] != vec[i + 1])
		{
			return true;
		}
	}
	return false;
}


int main()
{
	while (true)
	{
		int N;
		cin >> N; // N - number of packets
		if (N != -1)
		{
			vector <int> packets; // packets - array of packets
			cin.get();
			for (int i = 0; i < N; ++i)
			{
				int temp; // temp - temporary variable for storing number of candies in one packet
				cin >> temp;
				cin.get();
				packets.push_back (temp);
			}
			if (sum (packets) % packets.size())
			{
				cout << -1 << endl;
			}
			else
			{
				int result = 0;
				int mean = sum (packets) / packets.size();
				sort (packets);
				for (int i = 0, j = packets.size() - 1; allDifferent (packets); ++i)
				{
					while (packets[i] != mean)
					{
						if (packets[j] > mean)
						{
							++packets[i];
							--packets[j];
							++result;
						}
						else
						{
							--j;
						}
					}
				}
				cout << result << endl;
			}
		}
		else
		{
			break;
		}
	}
	return 0;
}
